### PR TITLE
Update codex

### PR DESCRIPTION
**Pull request by johnrc90**
Fix bugs and update new endpoints

Change the `notebooks/.env` to local neomaril and test the notebooks and the docs.

### CODE REVIEW

File: docs/source/deploying.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        df['proba'] = model.predict_proba(X)[:,1]
+        df['pred'] = model.predict(X)
+
+        output = data_path+'/output.csv'
+
+        df.to_csv(output, index=False)
+
+        return output
+
+The first parameter is now also a path for the data. We have different path parameter because each async model execution is saved in a different place. And the files uploaded when deploying the model are kept the same everytime.
```
Comment (ManuDiasCruz) [inline docs/source/deploying.rst:52]: Change "everytime" by "every time"

File: docs/source/deploying.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+If you want to keep your code more dinamic (and dont want to enforce a file name pattern) you can use the `inputFileName` env variable, that will be same as the filename uploaded for that execution.
```
Comment (ManuDiasCruz) [inline docs/source/deploying.rst:53]: Change "dinamic" by "dynamic" and "dont" by "don't"

File: docs/source/deploying.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    #                           env=PATH+'.env'  #  File for env variables (this will be encrypted in the server)
+    #                           extra_files=[PATH+'utils.py'], # List with extra files paths that should be uploaded along (they will be all in the same folder)
+                                python_version='3.9', # Can be 3.7 to 3.10
+                                operation="Sync", # Can be Sync or Async
+                                group='datarisk' # Model group (create one using the client)
+                                )
+
+
+
+As you can see deploying a model already trained in Neomaril requires less information (the AutoML models requrire only 2 parameters).
```
Comment (ManuDiasCruz) [inline docs/source/deploying.rst:103]: Inside the parenthesis, change "requrire" by "require"

File: docs/source/deploying.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    execution = async_model.predict(PATH+'input.csv')
+    # >>> 2023-05-26 12:04:14.714 | INFO     | neomaril_codex.model:predict:344 - Execution 5 started. Use the id to check its status.
+
+
+Sync models return a dictionary and async models return a :py:class:`neomaril_codex.base.NeomarilExecution` that you can use to check the status and download the result similiar to the training execution.
+
+To use the models you need a `group token`, that is generated when creating the group (check :ref:`connecting_to_neomaril:creating a group`). You can add this token in the NEOMARIL_GROUP_TOKEN env variable, use the :py:meth:`neomaril_codex.model.NeomarilModel.set_token` method or add in each :py:meth:`neomaril_codex.model.NeomarilModel.predict` call.
+
+
+Most of the time you might need to used your model outside a python enviroment, sharing it through a REST API.
```
Comment (ManuDiasCruz) [inline docs/source/deploying.rst:129]: Change "enviroment" by "environment"

File: docs/source/deploying.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+You can call the :py:attr:`neomaril_codex.model.NeomarilModel.docs` attribute to share a OpenAPI Swagger page, or use the :py:meth:`neomaril_codex.model.NeomarilModel.generate_predict_code` method to create a sample request code to your model. 
+
+
+Monitoring your model
+---------------------
+
+Model monitoring means keeping track with how the model is being used in production so we can update the model if it start making bad predictions.
+
+For now Neomaril only does indirect monitoring, that means following the input of the model in production and checking if is close to the data presented to the model in training.
+So when configurating the monitoring we need to know which training generated that model and what features are relevant to monitoring the model.
```
Comment (ManuDiasCruz) [inline docs/source/deploying.rst:139]: Change "configurating" by "configure"

File: docs/source/training_guide.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+Also, the training module is connected to the deploying and monitoring module, so using the training in Neomaril saves time in the next steps.
+
+
+Training types
+---------------
+
+First we need to create a training experiment. We aggregate multiple train runs into one training experiment. Each train run can eventually became a deployed model or not.
+
+**Custom:** Is when the user will define the whole training code and define the training enviroment.
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:14]: Change "enviroment" by "environment"

File: docs/source/training_guide.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+        results = pd.DataFrame({"pred": pipe.predict(X), "proba": pipe.predict_proba(X)[:,1]})
+        results.proba.hist().get_figure().savefig(base_path+'/probas.png', format='png')
+        
+        return {"X_train": X, "y_train": y, "model_output": results, "pipeline": pipe, 'extras': [base_path+'/probas.png'],
+                "metrics": {"auc": auc.mean(), "f1_score": f_score.mean()}}
+
+
+The only parameter on the function is the path for the data file. This way we can execute it when the files are uploaded to Neomaril.
+In the custom training experiment you can do whatever you want, test multiple algorithms, optimize hyperparameters, validate on multiple segments of the data.
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:61]: Change "hyperparameters" by "hyper parameters"

File: docs/source/training_guide.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+The important thing is the return of the function, where we get information about the final model of this version so we can log it. The return must be a dictionary with the following keys:
+
+- `X_train`: The dataframe tha will be used to fit the model.
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:64]: Change "tha" by "that"

File: docs/source/training_guide.rst [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+    run2 = training.run_training('First test', # Run name
+                                PATH+'dados.csv', # Path to the file with training data
+                                conf_dict=PATH+'conf.json', # Path of the configuration file
+                                wait_complete=True
+    )
+
+
+
+Checking the exectuion results
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:112]: Change "exectuion" by "execution"

File: docs/source/training_guide.rst [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+------------------------------
+
+The return of the :py:meth:`neomaril_codex.training.NeomarilTrainingExperiment.run_training` is a :py:class:`neomaril_codex.training.NeomarilTrainingExecution` instace
+With this class we can follow the assyncronous execution of that experiment version and check information on it.
```
Comment (ManuDiasCruz) [inline docs/source/training_guide.rst:116]: Change "assyncronous" by "asynchronous"

**Review by ManuDiasCruz — CHANGES_REQUESTED** [review]
I think we should deliver it after the update at GitHub neomaril repo to get the changes went from there as I talked to you.

Moreover I get some spell checker, that are not properly orthographic error, and let a suggestion list here (most of them I know that we in general use this abbreviation or short writings, but as it is a documentation you can consider which is the best thing to do):

Change "entrypoint" by "entry point"
Change "env" by "environment"
Change "enviroment" by "environment"
Change "dataframe" by "data frame"

**Comment by johnrc90** [comment]
> I think we should deliver it after the update at GitHub neomaril repo to get the changes went from there as I talked to you.
> 
> Moreover I get some spell checker, that are not properly orthographic error, and let a suggestion list here (most of them I know that we in general use this abbreviation or short writings, but as it is a documentation you can consider which is the best thing to do):
> 
> Change "entrypoint" by "entry point" Change "env" by "environment" Change "enviroment" by "environment" Change "dataframe" by "data frame"

Fixed typos here https://github.com/datarisk-io/mlops-neomaril-codex/pull/16/commits/ddb573dae85fcfc82ed910caf45523d6382a3e1f

I will wait to merge it until we have the changes in staging.

**Pull request merged**